### PR TITLE
HBW-376 Add tooltip message translation for string field

### PR DESCRIPTION
**Pull request by StasShk**

### CODE REVIEW

File: config/locales/bp/ru.new_customer.yml [hunk 1]
```diff
@@ -33,17 +43,25 @@ ru:
...
-      homsOrderDataCustomerEmail: Email
+      homsOrderDataCustomerName:
+        label: Имя
+      homsOrderDataCustomerSurname:
+        label: Фамилия
+      homsOrderDataCustomerPhone:
+        label: Телефон
+      homsOrderDataCustomerEmail:
+        label: Email
       homsOrderDataInstallDate: Дата подключения
```
Comment (TimAle) [inline config/locales/bp/ru.new_customer.yml:54]: It's actually confusing: for some types of fields yous should use `label`, for other types you shouldn't. Make it consistent, please.

File: hbw/app/javascript/packs/hbw/components/form/string.js.jsx [hunk 2]
```diff
@@ -87,9 +87,14 @@ modulejs.define('HBWFormString', ['React'], (React) => {
     }
 
     componentDidMount () {
+      const { task } = this.props;
+      const tooltip = this.context.translateBP(`${task.process_key}.${task.key}.${this.props.name}.message`,
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/form/string.js.jsx:91]: it's "title", "tooltip" is already occupied by `this.tooltip`

File: spec/hbw/features/bp_form/translation_spec.rb [hunk 3]
```diff
@@ -81,6 +87,10 @@
       expect(page).to     have_content 'submitSelectButton1'
       expect(page).to     have_content 'submitSelectButton2'
 
+      fill_in 'stringRegexpField', with: '123dfgdfg'
+
+      expect(page).to have_content 'Field does not match regex'
```
Comment (TimAle) [inline spec/hbw/features/bp_form/translation_spec.rb:92]: p

**Review by TimAle — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,18 +1,24 @@
 v2.8.0 [unreleased]
 -------------------
 ### Breaking changes
+- [#665](https://github.com/hydra-billing/homs/pull/665) Add tooltip message translation for string field
```
Comment (TimAle) [inline CHANGELOG.md:4]: a dot at the end, please

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,18 +1,24 @@
...
 - [#666](https://github.com/hydra-billing/homs/pull/666) Add translation for file upload field.
-  - BP localization config structure was changed:
+
+- BP localization config structure was changed:
 
   before:
    ```
+   stringField: Field label
    fileUploadField: Field label
```
Comment (TimAle) [inline CHANGELOG.md:12]: looks like you changed these two field types only, but it's not true

File: hbw/app/javascript/packs/hbw/components/form/string.js.jsx [hunk 6]
```diff
@@ -87,9 +87,14 @@ modulejs.define('HBWFormString', ['React'], (React) => {
     }
 
     componentDidMount () {
+      const { task } = this.props;
+      const tooltipMessage = this.context.translateBP(`${task.process_key}.${task.key}.${this.props.name}.message`,
+        {},
+        this.props.params.message || this.context.translate('errors.field_is_required'));
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/form/string.js.jsx:93]: prefetch `name` and `params` from props at line 90 please, keep the style consistent

**Review by TimAle — CHANGES_REQUESTED** [review]

**Comment by StasShk** [comment]
https://wiki.hydra-billing.ru/pages/viewpage.action?pageId=71368790

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,18 +1,25 @@
 v2.8.0 [unreleased]
 -------------------
 ### Breaking changes
+- [#665](https://github.com/hydra-billing/homs/pull/665) Add tooltip message translation for string field.
 - [#666](https://github.com/hydra-billing/homs/pull/666) Add translation for file upload field.
-  - BP localization config structure was changed:
 
+- BP localization config structure was changed. Now input labels for the next field types should be placed in key "Label": Checkbox, Datetime, FileList, FileUpload, Select, SelectTable, String, SubmitButton and Text.
+  String filed had additional key "message" for tooltip message, file_upload has input_text and browse_link_text keys.
```
Comment (TimAle) [inline CHANGELOG.md:8]:
```suggestion
- BP localization config structure was changed. Now input labels for the next field types should be placed in key "label": `checkbox`, `datetime`, `file_list`, `file_upload`, `select`, `select_table`, `string`, `submit_button` and `text`.
  `string` field has additional key "message" for tooltip message, `file_upload` has `input_text` and `browse_link_text` keys.
```

**Review by TimAle — CHANGES_REQUESTED** [review]

**Review by TimAle — APPROVED** [review]

**Pull request merged**